std/uuid.d

```diff
@@ -1058,18 +1058,21 @@ unittest
  *      randomGen = uniform RNG
  * See_Also: $(XREF random, isUniformRNG)
  */
-@trusted UUID randomUUID(RNG)(ref RNG randomGen) if(isIntegral!(typeof(RNG.front)))
+UUID randomUUID(RNG)(ref RNG randomGen) if(isIntegral!(typeof(RNG.front)))
 {
     import std.random : isUniformRNG;
     static assert(isUniformRNG!RNG, "randomGen must be an uniform RNG");
-    enum size_t elemSize = typeof(RNG.front).sizeof;
+
+    alias E = ElementEncodingType!RNG;
+    enum size_t elemSize = E.sizeof;
     static assert(elemSize <= 16);
+    static assert(16 % elemSize == 0);
+
     UUID u;
-    for(size_t i; i < 16; i += elemSize)
+    for (size_t i = 0 ; i < 16 ; i += elemSize)
     {
         randomGen.popFront();
-        immutable randomValue = randomGen.front;
-        u.data[i .. i + elemSize] = (*cast(ubyte[elemSize]*)&randomValue)[];
+        unsafeTrustingArrayElementReinterpret!E(u.data, i) = randomGen.front;
```

Comment:
Yes, the constraints don't check it; but that was actually removed in favor of a static assert (and isUniformPRNG, I believe, models infinite input). This was changed (I believe) to avoid un-needed dependencies.

I'd change the front/popFront thing, but I think that would break unittests, and I have not been able to rebuild dmd locally to test yet ...

Plus, I know there is at least 1 unittest that says MT19997 *MUST* map to a specific UUID, so changing anything would definitly break that...